src/main/java/uk/ac/ebi/eva/commons/core/models/VariantSource.java

```diff
@@ -76,7 +76,9 @@ public VariantSource(String fileId, String fileName, String studyId, String stud
         this.date = date;
         this.samplesPosition = samplesPosition;
         this.metadata = metadata;
-        this.stats = stats;
+        if (stats != null) {
+            this.stats = new VariantGlobalStats(stats);
```

Comment:
After testing this change with eva-ws it seems like it needs to be a concrete type when being deserialised.